File: backend/app/schemas/reservation.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class ReservationBase(BaseModel):
    evenement_id: int = Field(..., description="ID de l'evenement concerne")
    categorie_billet_id: int = Field(..., description="ID de la categorie de billet choisie (VIP, Simple, etc.)")


class ReservationCreate(ReservationBase):
    pass


class ReservationResponse(ReservationBase):
    id: int = Field(..., description="Unique ID de la reservation")
    statut: str
    participant_id: int
    date_reservation: datetime

    class Config:
        from_attributes = True


class PaiementConfirmeResponse(BaseModel):
    reservation_id: int
    statut_reservation: str
    montant_paye: float
    numero_billet: str
    qr_code: str

class ParticipantSummary(BaseModel):
    id: int
    nom: str
    prenom: str
    email: str

    class Config:
        from_attributes = True


class CategorieBilletSummary(BaseModel):
    id: int
    nom: str
    prix: float

    class Config:
        from_attributes = True


class ReservationDetailResponse(BaseModel):
    id: int
    statut: str
    date_reservation: datetime
    evenement_id: int
    participant: ParticipantSummary
    categorie_billet: CategorieBilletSummary

    class Config:
        from_attributes = True  

class ParticipantOrganisateurResponse(BaseModel):
    id: int
    nom: str
    prenom: str
    email: str
    telephone: Optional[str] = None
    nb_reservations: int
    montant_total_depense: float
    derniere_reservation: datetime

    class Config:
        from_attributes = True

class PaiementOrganisateurResponse(BaseModel):
    id: int
    montant: float
    statut_paiement: str
    mode_paiement: str
    date_paiement: Optional[datetime] = None
    evenement_titre: str
    participant_nom: str
    participant_prenom: str

    class Config:
        from_attributes = True

File: front-office/src/types/reservation.ts
export interface CategorieBillet {
  id: number
  nom: string
  prix: number
  quantite_disponible: number | null
  evenement_id: number
}

export interface Reservation {
  id: number
  evenement_id: number
  categorie_billet_id: number
  statut: string
  participant_id: number
  date_reservation: string
}

export interface PaiementConfirme {
  reservation_id: number
  statut_reservation: string
  montant_paye: number
  numero_billet: string
  qr_code: string
}

export interface Billet {
  id: number
  numero_billet: string
  qr_code: string
  is_used: boolean
  date_emission: string
  reservation_id: number
}

export interface ParticipantSummary {
  id: number
  nom: string
  prenom: string
  email: string
}

export interface CategorieBilletSummary {
  id: number
  nom: string
  prix: number
}

export interface ReservationDetail {
  id: number
  statut: string
  date_reservation: string
  evenement_id: number
  participant: ParticipantSummary
  categorie_billet: CategorieBilletSummary
}

export interface ParticipantOrganisateur {
  id: number
  nom: string
  prenom: string
  email: string
  telephone: string | null
  nb_reservations: number
  montant_total_depense: number
  derniere_reservation: string
}

export interface PaiementOrganisateur {
  id: number
  montant: number
  statut_paiement: string
  mode_paiement: string
  date_paiement: string | null
  evenement_titre: string
  participant_nom: string
  participant_prenom: string
}